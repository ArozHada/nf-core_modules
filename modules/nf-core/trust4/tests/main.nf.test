nextflow_process {

    name "Test Process TRUST4"
    script "../main.nf"
    process "TRUST4"

    tag "modules"
    tag "modules_nfcore"
    tag "trust4"

    test("sarscov2 - single_end") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [], // bam
                    [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)] // reads
                ])
                input[1] = Channel.of([file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)]) // fasta
                input[2] = [] // ref
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.airr_tsv,
                                process.out.fq,
                                process.out.fasta,
                                process.out.out,
                                process.out.report_tsv,
                                process.out.tsv,
                                process.out.versions,
                                ).match() }
            )
        }

    }

    test("sarscov2 - paired_end") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [], // bam
                    [file(params.test_data['sarscov2']['illumina']['test2_1_fastq_gz'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test2_2_fastq_gz'], checkIfExists: true)] // reads
                ])
                input[1] = Channel.of([file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)]) // fasta
                input[2] = [] // ref
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.airr_tsv,
                                process.out.fq,
                                process.out.fasta,
                                process.out.out,
                                process.out.report_tsv,
                                process.out.tsv,
                                process.out.versions,
                                ).match() }
            )
        }

    }

    test("sarscov2 - single_end - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [],
                    [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)]
                ])
                input[1] = Channel.of([file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)])
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.airr_tsv,
                                process.out.fq,
                                process.out.fasta,
                                process.out.out,
                                process.out.report_tsv,
                                process.out.tsv,
                                process.out.versions,
                                ).match() }
            )
        }

    }

}
