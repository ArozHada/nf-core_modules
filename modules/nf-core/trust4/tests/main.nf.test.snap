{
    "sarscov2 - single_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_airr.tsv:md5,572c69a08a82acc5ff9a2bf95488b0a4"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_toassemble.fq:md5,bd7ab2f9ea96290e58e868e4aabccf1f"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_annot.fa:md5,4b07aefcac3763f7f3d242d9371060df",
                        "test_assembled_reads.fa:md5,2c9601da894105f2fe6c068659214db3"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_cdr3.out",
                        "test_final.out:md5,db4a600993437157ed34be3fae73be25",
                        "test_raw.out:md5,db4a600993437157ed34be3fae73be25"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_report.tsv:md5,51623a22f060007a69207b1ebdf3a7cd"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_airr.tsv:md5,572c69a08a82acc5ff9a2bf95488b0a4",
                        "test_airr_align.tsv",
                        "test_report.tsv:md5,51623a22f060007a69207b1ebdf3a7cd"
                    ]
                ]
            ],
            [
                "versions.yml:md5,ef2e18f5d083f09211fb613305df93e8"
            ]
        ],
        "timestamp": "2024-01-22T16:53:18.214523"
    },
    "sarscov2 - single_end - stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_airr.tsv"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_toassemble.fq"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_annot.fa",
                        "test_assembled_reads.fa"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_cdr3.out",
                        "test_final.out",
                        "test_raw.out"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test_report.tsv"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    [
                        "test_airr.tsv",
                        "test_airr_align.tsv",
                        "test_report.tsv"
                    ]
                ]
            ],
            [
                "versions.yml:md5,ef2e18f5d083f09211fb613305df93e8"
            ]
        ],
        "timestamp": "2024-01-22T16:53:33.60176"
    },
    "sarscov2 - paired_end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_airr.tsv:md5,572c69a08a82acc5ff9a2bf95488b0a4"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_toassemble_1.fq:md5,bd7ab2f9ea96290e58e868e4aabccf1f",
                        "test_toassemble_2.fq:md5,0211588305d478e10879fb35b93e13d5"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_annot.fa:md5,4207f198195fa19b505ffa67feba776a",
                        "test_assembled_reads.fa:md5,60f27ccec04b5b1826ecf1fd87fd4ad6"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_cdr3.out",
                        "test_final.out:md5,8794c2e6c2306b589a3a5b050f9bd5df",
                        "test_raw.out:md5,5127266cf0810c970b90727600de50a3"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_report.tsv:md5,51623a22f060007a69207b1ebdf3a7cd"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_airr.tsv:md5,572c69a08a82acc5ff9a2bf95488b0a4",
                        "test_airr_align.tsv",
                        "test_report.tsv:md5,51623a22f060007a69207b1ebdf3a7cd"
                    ]
                ]
            ],
            [
                "versions.yml:md5,ef2e18f5d083f09211fb613305df93e8"
            ]
        ],
        "timestamp": "2024-01-22T16:53:26.63073"
    }
}